src/ImageProcessor.Web.Plugins.PostProcessor/PostProcessor.cs

```diff
@@ -41,116 +41,142 @@ internal static class PostProcessor
+                {
+                    ImageProcessorBootstrapper.Instance.Logger.Log(typeof(PostProcessor), $"Unable to post process image for request {context.Request.Unvalidated.Url} within {postProcessorBootstrapper.Timeout}ms. Original image returned.");
+                }
+                catch (Exception ex)
                 {
-                    // Ensure files exist, are not read only, and delete
-                    if (sourceFile != null && File.Exists(sourceFile))
+                    // Some security policies don't allow execution of programs in this way
+                    ImageProcessorBootstrapper.Instance.Logger.Log(typeof(PostProcessor), ex.Message);
+                }
+                finally
+                {
+                    // Set position back to begin
+                    stream.Position = 0;
+
+                    // Always cleanup files
+                    if (sourceFile != null)
                     {
-                        File.SetAttributes(sourceFile, FileAttributes.Normal);
-                        File.Delete(sourceFile);
+                        try
+                        {
+                            var sourceFileInfo = new FileInfo(sourceFile);
+                            if (sourceFileInfo.Exists)
+                            {
+                                sourceFileInfo.IsReadOnly = false;
+                                sourceFileInfo.Delete();
+                            }
+                        }
+                        catch
+                        {
+                            // Normally a no no, but logging would be excessive + temp files get cleaned up eventually
+                        }
                     }
 
-                    if (destinationFile != null && File.Exists(destinationFile))
+                    if (destinationFile != null)
                     {
-                        File.SetAttributes(destinationFile, FileAttributes.Normal);
-                        File.Delete(destinationFile);
+                        try
+                        {
+                            var destinationFileInfo = new FileInfo(destinationFile);
+                            if (destinationFileInfo.Exists)
+                            {
+                                destinationFileInfo.IsReadOnly = false;
+                                destinationFileInfo.Delete();
+                            }
+                        }
+                        catch
+                        {
+                            // Normally a no no, but logging would be excessive + temp files get cleaned up eventually
+                        }
                     }
                 }
-                catch
-                {
-                    // Normally a no no, but logging would be excessive + temp files get cleaned up eventually.
-                }
             }
 
             // ALways return stream (even if it's not optimized)
-            stream.Position = 0;
```

Comment:
The stream is only altered within the try-catch-finally, so setting the position back to the beginning only has to be done in the finally-statement.